File: typescript/src/worker.ts
import { NativeConnection, Worker } from '@temporalio/worker';

import { createActivities } from './activities';
import { TASK_QUEUE_NAME } from './shared';

async function run() {
  const connection = await NativeConnection.connect({
    address: process.env.TEMPORAL_ADDRESS,
    tls: process.env.TEMPORAL_TLS === 'true',
    apiKey: process.env.TEMPORAL_API_KEY,
  });

  try {
    // Create the worker with the task queue "hackathon"
    const worker = await Worker.create({
      connection,
      namespace: process.env.TEMPORAL_NAMESPACE,
      taskQueue: TASK_QUEUE_NAME,
      // Workflows are registered using a path as they run in a separate JS context.
      workflowsPath: require.resolve('./workflows'),
      // Register the activities - you may need to inject dependencies in here
      activities: createActivities(),
    });

    await worker.run();
  } finally {
    // Close the connection once the worker has stopped
    await connection.close();
  }
}

run().catch((err) => {
  console.error(err);
  process.exit(1);
});

File: python/src/worker.py
import asyncio
from temporalio.client import Client
from temporalio.worker import Worker
from temporalio.envconfig import ClientConfigProfile
from workflows import HelloWorldWorkflow
from activities import sayName
import shared


async def main():
    default_profile = ClientConfigProfile.load()
    connect_config = default_profile.to_client_connect_config()

    client = await Client.connect(**connect_config)
    worker = Worker(
        client,
        task_queue=shared.TASK_QUEUE_NAME,
        workflows=[HelloWorldWorkflow],
        activities=[sayName],
    )

    print("Worker started.")
    await worker.run()


if __name__ == "__main__":
    asyncio.run(main())
